src/cli/commands/lint.ts

```diff
@@ -169,39 +160,14 @@ async function lint(name: string, flags: ILintConfig, command: Lint, rules?: Rul
     return false;
   });
 
-  if (parseInt(spec.data.swagger) === 2) {
-    command.log('Adding OpenAPI 2.0 (Swagger) functions');
-    spectral.addFunctions(oas2Functions());
-  } else if (parseInt(spec.data.openapi) === 3) {
-    command.log('Adding OpenAPI 3.x functions');
-    spectral.addFunctions(oas3Functions());
-  }
-
-  if (rules) {
-    if (flags.verbose) {
-      command.log(`Found ${Object.keys(rules).length} rules`);
-    }
-  } else {
-    if (flags.verbose) {
-      command.log('No rules loaded, attempting to detect document type');
-    }
-
-    rules = Object.assign(
-      {},
-      await tryReadOrLog(command, async () => await oas2Rules()),
-      await tryReadOrLog(command, async () => await oas3Rules()),
-    );
+  if (rulesets && flags.verbose) {
+    command.log(`Found ${Object.keys(rulesets).length} rulesets`);
   }
 
   if (flags.skipRule) {
-    rules = skipRules({ ...rules }, flags, command);
-  }
-  if (!rules) {
-    throw new Error('No rules provided, and document type does not have any default rules, so lint has nothing to do');
+    skipRules(spectral.rules, flags, command);
```

Comment:
Is this function mutating the public property? Is that the way we want to do this?

Reply:
You are correct, it mutates the property.
Should be harmless in this case, but actually I think it'd be nice if we provided a `removeRule` method or something to Spectral.

BTW, review https://github.com/stoplightio/spectral/pull/430 first - #460 PR is based on it.